ocs_ci/ocs/resources/pod.py

```diff
@@ -1056,3 +1060,57 @@ def upload(pod_name, localpath, remotepath):
     """
     cmd = f"oc cp {os.path.expanduser(localpath)} {pod_name}:{remotepath}"
     run_cmd(cmd)
+
+
+def verify_pods_upgraded(old_images, selector, count=1, timeout=600):
+    """
+    Verify that all pods do not have old image.
+
+    Args:
+       old_images (set): Set with old images.
+       selector (str): Selector (e.g. app=ocs-osd)
+       count (int): Number of resources for selector.
+       timeout (int): Timeout in seconds to wait for pods to be upgraded.
+
+    Raises:
+        TimeoutException: If the pods didn't get upgraded till the timeout.
+
+    """
+
+    namespace = config.ENV_DATA['cluster_namespace']
+    pod = OCP(
+        kind=constants.POD, namespace=namespace,
+    )
+    info_message = (
+        f"Waiting for {count} pods with selector: {selector} to be running "
+        f"and upgraded."
+    )
+    logger.info(info_message)
+    start_time = time.time()
+    selector_label, selector_value = selector.split('=')
+    while True:
+        pod_count = 0
+        try:
+            pods = get_all_pods(namespace, [selector_value], selector_label)
+            pods_len = len(pods)
+            logger.info(f"Found {pods_len} pod(s) for selector: {selector}")
+            if pods_len != count:
+                logger.warning(
+                    f"Number of found pods {pods_len} is not as expected: "
+                    f"{count}"
+                )
+            for pod in pods:
+                verify_images_upgraded(old_images, pod.get())
+                pod_count += 1
+        except CommandFailed as ex:
+            logger.warning(
+                f"Failed when getting pods with selector {selector}."
+                f"Error: {ex}"
+            )
+        except NonUpgradedImagesFoundError as ex:
+            logger.warning(ex)
+        timeout_reached(start_time, timeout, info_message)
```

Comment:
s/timeout_reached/check_timeout_reached/,  Why not catch exception and fail here for better readability?